nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py

```diff
@@ -570,6 +570,71 @@ def process_micro_batch(self, micro_batch):
 
         return tokens_enc, tokens_dec, loss_mask, labels, enc_mask, dec_mask
 
+    def _process_global_batch_without_megatron_batch_sampler(self, global_batch, tokenizer=None):
```

Comment:
Wouldn't it better to do this as part of dataloader as discussed? 
If that's upcoming in an upcoming/broader PR then looks good

Reply:
Yeah that refactor will be part of an upcoming PR. Other fine-tuning models also need to use something like this for now.